documentation/versioning.md

```diff
@@ -0,0 +1,118 @@
+    - Ideally, we’d get rid of the resource-manager folder’s RPNamespace subfolder as this extra level in the hierarchy serves no purpose and is just confusing, but this can be done in a later phase.
+2.	In the resource-manager folder, if it's RPaaS service, there must be just 1 readme.md file, and 1+ service name folders under resource-manager/RPnamespace folder, but the resource types within each services folder must not duplicate except for operations, locations, checkNameAvailability etc. Which is required by RPSaaS for HTTP request schema validation.
+3.	In the data-plane folder there must be no files and at least 1 service name folder.
+4.	The ServiceName folder must have only readme.*, tspconfig.yaml and *.tsp files in it. 
+    - In the readme.md file, the latest package tag value (which represents a service version) must refer to swagger files that are all in the exact same folder. Also, if the tag value is suffixed with “-preview” then all the swagger paths must refer to files in the preview subfolder; else all swagger paths must refer to files in the stable subfolder.
+    - NOTE: If 2+ services want to “share” some TypeSpec/swagger files, these files must be explicitly copied into a consuming service’s folder on its desired timeline. We do not allow multiple services to reference a single copy of a service contract because it is too dangerous to allow one service to change a shared contract, forcing other services to pick up this shared contract instantly, especially if the shared contract introduces breaking changes. Each service must be in control of its own destiny.
+5.	The ServiceName folder must have examples, preview and stable subfolders. Additional subfolders are allowed to help organize *.tsp files. The rules should ensure that only the tspconfig.yaml  and *.tsp files are in the ServiceName folder.  
+6.	The examples folder should only have a subfolder for each api-version and each api-version folder can contain only example *.json files for this api-version.
+7.	The preview and stable folders must only have subfolders whose name match this format: YYYY-MM-DD-preview in the preview folder and YYYY-MM-DD in the stable folder. All the files in the YYYY-MM-DD(-preview) folders must be json (swagger) files.
+8.	Each YYYY-MM-DD(-preview) folder must have an examples folder containing only example *.json files for this api-version.
+9.	It is illegal for a preview & stable version to share the same date. For example, if there is a 2024-04-01-preview, then there can never be a 2024-04-01 stable. It would be great for the repo tooling to detect this and prevent merging of the PR. 
+10. It is not recommended if different service name folder share the same apiVersion when service adds a new version. 
+11.	Each service name folder must duplicate the special “Operations operation” (see RPC docs, real-life docs and folder example). 
```

Comment:
I modify this part based on our latest practice.

Reply:
Here I think we mean the 'list operations'  api that is required by ARM?

Reply:
Yes, in Jeffrey's docx, it's still the old version where there's to have a presudo operations service folder, but due to our latest practice and confirm with ARM team, we recommend service team to duplicate the operations list api in each service folder.

Reply:
Do we need to maintain two copies, I thought we are changing the source of truth to this one? If necessary I will make the changes back to his doc.

Reply:
I think just proceeding with the Wiki is OK.
The only downside I see is that the link to the word doc has been given out a LOT over the past few years and will probably continue to be forwarded to people and this will not be out of date.

@qiaozha: Could you update my Word doc at the very top in a big font with yellow highlight that the doc has been replaced with the Wiki and put the link to the Wiki in the Word doc. Also add that the doc will not be kept up to date - the Wiki is the new source of truth.

Thank you.

Reply:
here's the wiki link https://github.com/Azure/azure-rest-api-specs/wiki/Azure-Service-Versioning-Guideline and the docx is also updated. Thanks!